forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -1677,6 +1678,10 @@ else if (property.equals("blocked")) {
             if (band == null || !band.getAttackers().contains(card)) {
                 return false;
             }
+        } else if (property.equals("mustAttack")) {
+            if (!card.isGoaded() && StaticAbilityMustAttack.entitiesMustAttack(card).isEmpty()) {
```

Comment:
The Problem is it needs to look back
If the Attacker gets goaded after being declared it doesnt matter